### PR TITLE
fixing return_embedding issue for time regression model

### PR DESCRIPTION
**Pull request by NiranjanChaudhari0929**
This branch is created to fix the issue of size when running with 'time' as label

### CODE REVIEW

File: src/lmcontrol/nn/clf.py [hunk 1]
```diff
@@ -59,7 +59,9 @@ def __init__(self, num_outputs=1, lr=0.01, step_size=2, gamma=0.1, planes=[8, 16
         
         self.backbone = ResNet(label_type=label_type, block=block, layers=layers, planes=planes, num_outputs=num_outputs, return_embeddings=return_embeddings)
         if label_type == 'time':
-            self.backbone = nn.Sequential(self.backbone, nn.Softplus(), nn.Flatten(start_dim=0))
+            self.backbone = nn.Sequential(self.backbone, nn.Softplus(), nn.Flatten(start_dim=1)) #change bac to 0 once done
```
Comment (ajtritt) [inline src/lmcontrol/nn/clf.py:62]:
```suggestion
            self.backbone = nn.Sequential(self.backbone, nn.Softplus(), nn.Flatten(start_dim=1))
```

File: src/lmcontrol/nn/clf.py [hunk 2]
```diff
@@ -391,13 +393,13 @@ def predict(argv=None):
...
     trainer = L.Trainer(devices=1, accelerator=accelerator)
-
+    
     logger.info("Running predictions")
     predictions = trainer.predict(model, predict_loader)
     predictions = torch.cat(predictions).numpy()
 
     if args.return_embeddings:
-        logger.info("No classifier mode: Saving embeddings")
+        logger.info("No regression mode: Saving embeddings")
```
Comment (ajtritt) [inline src/lmcontrol/nn/clf.py:402]:
```suggestion
        logger.info("Saving embeddings")
```

File: src/lmcontrol/nn/clf.py [hunk 3]
```diff
@@ -454,6 +456,7 @@ def predict(argv=None):
             out_data[k + "_labels"] = np.asarray(dset.labels[:, i])
 
     np.savez(args.output_npz, **out_data)
+    #np.savez(args.output_npz, **{key: np.array(val, dtype=object) for key, val in out_data.items()})
```
Comment (ajtritt) [inline src/lmcontrol/nn/clf.py:459]:
```suggestion
```

File: src/lmcontrol/nn/dataset.py [hunk 4]
```diff
@@ -99,9 +99,9 @@ def __init__(self, npzs, use_masks=False, return_labels=False, logger=None, tran
             self.label_type = list()
             self.label_classes = list()
             for k in metadata:
-                if not return_embeddings:           
-                    if k not in label_type:
-                        continue
+                #if not return_embeddings:
```
Comment (ajtritt) [inline src/lmcontrol/nn/dataset.py:102]:
```suggestion
```

**Review by ajtritt — CHANGES_REQUESTED** [review]

File: src/lmcontrol/nn/clf.py [hunk 5]
```diff
@@ -59,7 +59,9 @@ def __init__(self, num_outputs=1, lr=0.01, step_size=2, gamma=0.1, planes=[8, 16
         
         self.backbone = ResNet(label_type=label_type, block=block, layers=layers, planes=planes, num_outputs=num_outputs, return_embeddings=return_embeddings)
         if label_type == 'time':
-            self.backbone = nn.Sequential(self.backbone, nn.Softplus(), nn.Flatten(start_dim=0))
+            self.backbone = nn.Sequential(self.backbone, nn.Softplus(), nn.Flatten(start_dim=1))
+        # if label_type == 'time' and return_embeddings:
+        #     self.backbone = nn.Sequential(self.backbone, nn.Softplus(), nn.Flatten(start_dim=1))
```
Comment (ajtritt) [inline src/lmcontrol/nn/clf.py:64]:
```suggestion
```

**Review by ajtritt — APPROVED** [review]
